static_src/stores/user_store.js

```diff
@@ -338,6 +355,36 @@ export class UserStore extends BaseStore {
     return role;
   }
 
+  addRoleToSavingArray(roleId) {
```

Comment:
So I feel liek this has created a whole new, complicated data structure to keep saving state when we could use additional data structure of of the array of users instead. I think I tried to explain this the other day but wasn't clear enough, sorry about that.

So we have a list of users in this store. on `USER_ROLES_DELETE` and `USER_ROLES_ADD`, just add a `saving: true` property to the user object with the guid of the user who's role is being deleted or added. Then on `USER_ROLES_DELETED`, `USER_ROLES_ADD` or an error condition, just set `saving: false` on that same user object. Now the data for whether a specific user is being saved is right on that user object.

This will work well with the UI because we'll probably put a loading indicator on each user object rather the the whole user list (Think if you have a long user list; you won't see the loading state at the top as your editing roles at the bottom). We may need a global saving state for actions that aren't tied to a specific user, like inviting a new user.